src/shogun/features/DenseFeatures.h

```diff
@@ -541,7 +542,7 @@ template<class ST> class CDenseFeatures: public CDotFeatures
 	 * offset, it throws an error. It then copies into the target matrix, starting
 	 * from base + (colum_offset * num_features) location.
 	 */
-	void copy_feature_matrix(SGMatrix<ST> target, index_t column_offset=0) const;
+	void copy_feature_matrix(SGMatrix<ST>& target, index_t column_offset=0) const;
```

Comment:
avoid copying stuff around

Reply:
I think these things should be removed eventually

Reply:
Actually might be good
But why not return the matrix?

Reply:
i guess because RVO is not guaranteed by the standard until c++17